### ISSUE TITLE
Unused modules with error messages are logged

### ISSUE THREAD

**Issue by thusoy** [saltstack/salt#21185]
The somewhat undocumented feature of returning a tuple from the `__virtual__` function in modules describing why the module could not be loaded causes error logging even if the module was not going to be used. I'm not sure how widely this feature is used, but at least the new ipmi and nacl modules use it, and cause error logging on develop.

**Comment by rallytime** [comment]
@thusoy Thanks for reporting this. Can you provide an example of the error logging you're seeing? We've made some changes to the loader recently that allows you to return the reason as to why a module wasn't loaded, as you've stated, but it definitely shouldn't be logging errors if the modules are not are being used. 

ping @cachedout and @jacksontj

**Comment by thusoy** [comment]
Sure, installing salt from git develop and running highstate:

```sh
...
[ERROR   ] Error loading module.ipmi: No module named pyghmi.ipmi
[ERROR   ] Error loading module.nacl: libnacl import error, perhaps missing python libnacl package
[ERROR   ] Error loading states.ipmi: No module named pyghmi.ipmi
...
```

**Comment by rallytime** [comment]
Awesome, thanks! Looks like @jacksontj is already working on a fix. :)

**Comment by jacksontj** [comment]
The commit I pushed foxes the only condition I could find. If you still see
a problem let me know

**Comment by thusoy** [comment]
Thanks, I'll give it a spin.

**Comment by thusoy** [comment]
@jacksontj With your patch I'm only seeing the error logged once, where previously it would be logged several times. So we're at least not reloading the modules *that often* anymore, but the error is still being logged when it is.

**Comment by jacksontj** [comment]
Is there a specific way to reproduce? I'm thinking we might want to change
that down to debug instead of error

**Comment by steverweber** [comment]
aggreed this message should be debug level or removed alltogeather.

I would like to see the error passed to the user when they try to used the module when its not aviable.

**Comment by thusoy** [comment]
From a clean VM I can reproduce by running `curl -sL https://bootstrap.saltstack.com | sudo sh -s -- git develop`, adding an empty top file:
```yaml
base:
  '*': []
```

and run `sudo salt-call state.apply -l warning --local`.
```shell
[ERROR   ] Error loading module.nacl: libnacl import error, perhaps missing python libnacl package
[ERROR   ] Error loading module.ipmi: No module named pyghmi.ipmi
[ERROR   ] Error loading states.ipmi: No module named pyghmi.ipmi
[ERROR   ] Received bad data when setting the match from the top file
local:
    ----------
    no_|-states_|-states_|-None:
        ----------
        __run_num__:
            0
        changes:
            ----------
        comment:
            No Top file or external nodes data matches found
        name:
            No States
        result:
            False
```
The last error message is expected, the first ones are not. Taking the level down to debug or info makes a lot of sense.

**Comment by thusoy** [comment]
Fixed in #21266 .

**Comment by rallytime** [comment]
Awesome! Thanks for following up here @thusoy and for the fixes @jacksontj!
